sway-ir/src/optimize/simplify_cfg.rs

```diff
@@ -13,17 +13,33 @@ use crate::{
     value::ValueDatum,
 };
 
-pub fn simplify_cfg(context: &mut Context, function: &Function) -> Result<bool, IrError> {
-    let mut modified = false;
-    loop {
-        if remove_dead_blocks(context, function)? {
-            modified = true;
-            continue;
+use std::collections::HashMap;
+
+type PredCounts = HashMap<Block, usize>;
+
+fn count_predecessors(context: &Context, function: &Function) -> PredCounts {
```

Comment:
We might want to adjust `Block::num_predecessors()` too, if this algorithms is better.

Reply:
> I have a feeling this would be better as a method on `Function` as it will probably be useful in other passes. 

Okay, I'll move it there.

> And there's probably no need to iterate through `successors()` to just count them, you could use `len()`.

I'm not counting the `successors()` here but for each of them, updating their predecessor count. So I'll need to iterate through them right? Or I misunderstood your comment.

Reply:
> We might want to adjust `Block::num_predecessors()` too, if this algorithms is better.

I've address all review comments except this one. I think we should do it separately if needed. At this point I don't know what is better for here.

Reply:
Yep, and yeah, I misread the 'counting successors' bit, sorry.  Probably shouldn't be reviewing PRs at night. :)  But here I am anyway...